cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -241,4 +241,19 @@ public Secret cruiseControlSecret() {
             isMaintenanceTimeWindowsSatisfied);
     }
 
+    @Override
+    protected String caCertThumbprintAnnotation() {
+        return "cluster-ca-thumbprint";
+    }
+
+    @SuppressWarnings("BooleanExpressionComplexity")
+    @Override
+    protected boolean isCaCertThumbprintChanged() {
+        // at least one Secret has a different cluster CA certificate thumbprint.
+        // it is useful when a renewal cluster CA certificate process needs to be recovered after an operator crash
+        return isCaCertThumbprintChanged(zkNodesSecret) || isCaCertThumbprintChanged(brokersSecret) ||
+                isCaCertThumbprintChanged(entityTopicOperatorSecret) || isCaCertThumbprintChanged(entityUserOperatorSecret) ||
+                isCaCertThumbprintChanged(kafkaExporterSecret) || isCaCertThumbprintChanged(cruiseControlSecret) ||
+                isCaCertThumbprintChanged(clusterOperatorSecret);
+    }
```

Comment:
I was thinking about putting the CA cert thumbprint annotation on the ZK and K pods (so on `StatefulSet` and through the `StrimziPodSet`) as it happens today for the CA cert generation. This could help the `KafkaAssemblyOperator#getReasonsToRestartPod` to check that even if `isCaCertThumbprintChanged` is true, we have to check specific pods annotation because them could be already updated so no need to roll.
It's also true that during the first roll about trusting the new CA, the thumbprint annotation is not updated but it happens on the second one on certificates generation. :-(